frontend/src/app/[locale]/formPrototype/[applicationId]/[formId]/page.tsx

```diff
@@ -31,31 +36,71 @@ interface formPageProps {
 }
 
 async function FormPage({ params }: formPageProps) {
-  const { formId } = await params;
+  const { applicationId, formId } = await params;
   let formData = {} as FormDetail;
+  let applicationData = {} as ApplicationDetail;
 
   try {
     const response = await getFormDetails(formId);
     if (response.status_code !== 200) {
-      throw new Error(
-        `Error retrieving competition details: ${JSON.stringify(response.errors)}`,
+      console.error(
+        `Error retrieving form details for formID (${formId})`,
+        response,
       );
+      return <TopLevelError />;
     }
     formData = response.data;
-  } catch (error) {
-    if (parseErrorStatus(error as ApiRequestError) === 404) {
+  } catch (e) {
+    console.error(
+      `Error retrieving application details for formId ${formId}:`,
+      e,
+    );
+    if (parseErrorStatus(e as ApiRequestError) === 404) {
+      return <NotFound />;
+    }
+    return <TopLevelError />;
```

Comment:
There isn't a good error boundary at the moment, so not throwing errors. If it doesn't exist in the API, you get a 404, otherwise just a page error. We can revisit after the demo, once we are really building something.